E712: False negative when multiple comparison in same line

Sample code:
```python
var = 1 if cond == True else -1 if cond == False else cond

if cond == False or cond == True:
    print(var)
```
When using pycodestyle 2.7.0
```
$ pycodestyle  mytest.py 
mytest.py:1:17: E712 comparison to True should be 'if cond is True:' or 'if cond:'
mytest.py:3:9: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
```
When multiple equality comparisons are present in the same line, only first occurrence is detected.

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -1393,8 +1393,10 @@ def comparison_to_singleton(logical_line, noqa):
     None was set to some other value.  The other value might have a type
     (such as a container) that could be false in a boolean context!
     """
-    match = not noqa and COMPARE_SINGLETON_REGEX.search(logical_line)
-    if match:
+    if noqa:
+        return
+
+    for match in COMPARE_SINGLETON_REGEX.finditer(logical_line):
         singleton = match.group(1) or match.group(3)
         same = (match.group(2) == '==')
 

--- a/testsuite/E71.py
+++ b/testsuite/E71.py
@@ -42,6 +42,8 @@ if res[1] == True:
 #: E712
 if res[1] != False:
     pass
+#: E712 E712
+var = 1 if cond == True else -1 if cond == False else cond
 
 #
 #: E713

